test/cppsuite/tests/basic_test.cxx

```diff
@@ -0,0 +1,155 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/connection_manager.h"
+#include "test_harness/thread_manager.h"
+#include "test_harness/util/api_const.h"
+#include "test_harness/util/logger.h"
+#include "test_harness/workload/random_generator.h"
+
+extern "C" {
+#include "wiredtiger.h"
+}
+
+using namespace test_harness;
+
+bool do_inserts = false;
+bool do_reads = false;
+
+void
+insert_op(WT_CURSOR *cursor, int key_size, int value_size)
+{
+    logger::log_msg(LOG_INFO, "called insert_op");
+
+    /* Insert random data. */
+    std::string key, value;
+    while (do_inserts) {
+        key = random_generator::instance().generate_random_string(key_size);
+        value = random_generator::instance().generate_random_string(value_size);
+        cursor->set_key(cursor, key.c_str());
+        cursor->set_value(cursor, value.c_str());
+        testutil_check(cursor->insert(cursor));
+    }
+}
+
+void
+read_op(WT_CURSOR *cursor, int key_size)
+{
+    logger::log_msg(LOG_INFO, "called read_op");
+
+    /* Read random data. */
+    std::string key;
+    // while (do_reads) {
```

Comment:
The insert operations fail when this is uncommented. Will submit a fix when the issue is resolved.